src/Microsoft.FeatureManagement/Targeting/ContextualTargetingFilter.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.FeatureManagement.FeatureFilters
+{
+    /// <summary>
+    /// A feature filter that can be used to activate features for targeted audiences.
+    /// </summary>
+    [FilterAlias(Alias)]
+    public class ContextualTargetingFilter : IContextualFeatureFilter<ITargetingContext>
+    {
+        private const string Alias = "Microsoft.Targeting";
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// Creates a targeting contextual feature filter.
+        /// </summary>
+        /// <param name="loggerFactory">A logger factory for creating loggers.</param>
+        public ContextualTargetingFilter(ILoggerFactory loggerFactory)
+        {
+            _logger = loggerFactory?.CreateLogger<ContextualTargetingFilter>() ?? throw new ArgumentNullException(nameof(loggerFactory));
+        }
+
+        /// <summary>
+        /// Performs a targeting evaluation using the provided <see cref="TargetingContext"/> to determine if a feature should be enabled.
+        /// </summary>
+        /// <param name="context">The feature evaluation context.</param>
+        /// <param name="targetingContext">The targeting context to use during targeting evaluation.</param>
+        /// <returns>True if the feature is enabled, false otherwise.</returns>
+        public Task<bool> EvaluateAsync(FeatureFilterEvaluationContext context, ITargetingContext targetingContext)
```

Comment:
>context [](start = 71, length = 7)

null checking?

Reply:
FeatureFilterEvaluationContext is guaranteed to not be null. The second parameter could be null, but I'm wondering if we should change it to disallow null.

Reply:
I'm going to go ahead and add null check for both of them even though the state machine won't pass null for the first parameter. The method is better off establishing it's contract explicitly. Good suggestion.